cypress/e2e/hub/remotes-access.cy.ts

```diff
@@ -0,0 +1,126 @@
+import { hubAPI } from '../../support/formatApiPathForHub';
+import { Remotes } from './constants';
+import { HubRemote } from '../../../frontend/hub/administration/remotes/Remotes';
+
+describe.skip('Remotes User Access tab', () => {
+  let remote: HubRemote;
+  before(() => {
+    cy.createHubRemote().then((createdRemote) => {
+      remote = createdRemote;
+    });
+  });
+
+  after(() => {
+    cy.deleteHubRemote(remote);
+  });
+
+  beforeEach(() => {
+    cy.navigateTo('hub', 'remotes');
+    cy.verifyPageTitle(Remotes.title);
+    cy.filterTableBySingleText(remote.name);
+    cy.clickTableRowLink('name', remote.name, { disableFilter: true });
+    cy.verifyPageTitle(remote.name);
+  });
+
+  function removeRoleFromListRow(roleName: string) {
+    cy.clickTableRowPinnedAction(roleName, 'remove-role', false);
+    cy.getModal().within(() => {
+      cy.get('#confirm').click();
+      cy.clickButton(/^Remove role/);
+      cy.contains(/^Success$/).should('be.visible');
+      cy.containsBy('button', /^Close$/).click();
+    });
+  }
+
+  it('create a new remote, from the user access tab assign a user and apply role(s) to the user of the remote', () => {
+    cy.intercept('POST', hubAPI`/_ui/v2/role_user_assignments/`).as('userRoleAssignment');
+    cy.createHubUser().then((hubUser) => {
+      cy.clickTab('User Access', true);
+      cy.getByDataCy('add-roles').click();
+      cy.verifyPageTitle('Add roles');
+
+      cy.getWizard().within(() => {
+        cy.contains('h1', 'Select user(s)').should('be.visible');
+        cy.selectTableRow(hubUser.username);
+        cy.clickButton(/^Next/);
+        cy.contains('h1', 'Select roles to apply').should('be.visible');
+        cy.filterTableByTextFilter('name', 'galaxy.collection_remote_owner', {
+          disableFilterSelection: true,
+        });
+        cy.selectTableRowByCheckbox('name', 'galaxy.collection_remote_owner', {
+          disableFilter: true,
+        });
+        cy.clickButton(/^Next/);
+        cy.contains('h1', 'Review').should('be.visible');
+        cy.verifyReviewStepWizardDetails('users', [hubUser.username], '1');
+        cy.verifyReviewStepWizardDetails(
+          'hubRoles',
+          ['galaxy.collection_remote_owner', 'Manage collection remotes.'],
+          '1'
+        );
+        cy.clickButton(/^Finish/);
+        cy.wait('@userRoleAssignment')
```

Comment:
Not sure what we're testing in the UI here. If we want to ensure roles are properly assigned, should we verify the redirect happens and that the main table shows the assigned roles?

Reply:
are you referring to the wait or this command?

1. If it is about `cy.wait`, I think it is necessary in places where it will take time for the operation/API call to be done before we make another assertion.

2. If it is about the `verifyReviewStepWizardDetails` this command verifies the review step details before the POST call happens, and the team/user gets added.
`cy.verifyReviewStepWizardDetails(
          'hubRoles',
          ['galaxy.collection_remote_owner', 'Manage collection remotes.'],
          '1'
        );`

3. 
> If we want to ensure roles are properly assigned, should we verify the redirect happens and that the main table shows the assigned roles

Searching and filtering the user is enough for the above. wait and review step commands are additional measures to make the test robust

`cy.selectTableRowByCheckbox('username', hubUser.username, {
        disableFilter: true,
      });`

@nixocio